ArcGISRuntimeSDKQt_CppSamples/Layers/ExportVectorTiles/ExportVectorTiles.cpp

```diff
@@ -0,0 +1,140 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#ifdef PCH_BUILD
+#include "pch.hpp"
+#endif // PCH_BUILD
+
+#include "ExportVectorTiles.h"
+
+#include "ArcGISVectorTiledLayer.h"
+#include "ExportVectorTilesTask.h"
+#include "ExportVectorTilesParameters.h"
+#include "GeometryEngine.h"
+#include "GraphicsOverlay.h"
+#include "Map.h"
+#include "MapQuickView.h"
+#include "SimpleLineSymbol.h"
+
+#include <QTemporaryDir>
+
+using namespace Esri::ArcGISRuntime;
+
+ExportVectorTiles::ExportVectorTiles(QObject* parent /* = nullptr */):
+  QObject(parent),
+  m_map(new Map(BasemapStyle::ArcGISStreetsNight, this))
+{
+}
+
+ExportVectorTiles::~ExportVectorTiles() = default;
+
+void ExportVectorTiles::init()
+{
+  // Register the map view for QML
+  qmlRegisterType<MapQuickView>("Esri.Samples", 1, 0, "MapView");
+  qmlRegisterType<ExportVectorTiles>("Esri.Samples", 1, 0, "ExportVectorTilesSample");
+}
+
+MapQuickView* ExportVectorTiles::mapView() const
+{
+  return m_mapView;
+}
+
+// Set the view (created in QML)
+void ExportVectorTiles::setMapView(MapQuickView* mapView)
+{
+  if (!mapView || mapView == m_mapView)
+    return;
+
+  m_mapView = mapView;
+  m_mapView->setMap(m_map);
+  m_mapView->setViewpoint(Viewpoint(34.049, -117.181, 1e4));
+
+  m_graphicsOverlay = new GraphicsOverlay(this);
+  m_mapView->graphicsOverlays()->append(m_graphicsOverlay);
+
+  // Create the graphic that will be used to show the export extent
+  m_exportAreaGraphic = new Graphic(this);
+  m_exportAreaGraphic->setSymbol(new SimpleLineSymbol(SimpleLineSymbolStyle::Solid, Qt::green, 3, this));
+
+  m_graphicsOverlay->graphics()->append(m_exportAreaGraphic);
+
+  emit mapViewChanged();
+}
+
+void ExportVectorTiles::startExport(double xSW, double ySW, double xNE, double yNE)
+{
+  if (m_map->basemap()->baseLayers()->first()->layerType() != LayerType::ArcGISVectorTiledLayer)
+    return;
+
+  // Get the first layer of the basemap baselayers as a vector tiled layer for export
+  ArcGISVectorTiledLayer* vectorTiledLayer = static_cast<ArcGISVectorTiledLayer*>(m_map->basemap()->baseLayers()->first());
+  ExportVectorTilesTask* exportTask = new ExportVectorTilesTask(vectorTiledLayer->url(), this);
+
+  // Create an envelope from the QML rectangle corners
+  const Point corner1 = m_mapView->screenToLocation(xSW, ySW);
+  const Point corner2 = m_mapView->screenToLocation(xNE, yNE);
+  const Envelope extent = Envelope(corner1, corner2);
+  const Envelope exportArea = GeometryEngine::project(extent, vectorTiledLayer->spatialReference());
```

Comment:
From @ldanzinger: 

Normalizing the geometry addresses this issue. I posted a message on the Waverly team channel to see if this is a bug or if it's expected.

```suggestion
 const Geometry exportArea = GeometryEngine::normalizeCentralMeridian(GeometryEngine::project(extent, vectorTiledLayer->spatialReference()));
```